### PR TITLE
CORE-8621 Refactor DiskResources module Part 5

### PR DESCRIPTION
**Pull request by ash3rz**
NOTE: This is a branch on top of another branch.  This PR's commits start at `CORE-8621 Refactor DiskResourceColumnModel size provider and cells into separate classes`

### CODE REVIEW

File: de-lib/src/main/java/org/iplantc/de/diskResource/client/events/selection/MetaTemplateInfoBtnSelected.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import com.google.gwt.event.shared.EventHandler;
+import com.google.gwt.event.shared.GwtEvent;
+import com.google.gwt.event.shared.HandlerRegistration;
+
+/**
+ * A GWT event that fires when the user hits the "info" button for a metadata template
+ */
+public class MetaTemplateInfoBtnSelected
+        extends GwtEvent<MetaTemplateInfoBtnSelected.MetaTemplateInfoBtnSelectedHandler> {
+    public static interface MetaTemplateInfoBtnSelectedHandler extends EventHandler {
```
Comment (tedgin) [inline de-lib/src/main/java/org/iplantc/de/diskResource/client/events/selection/MetaTemplateInfoBtnSelected.java]: Aren't all interfaces static?

File: de-lib/src/main/java/org/iplantc/de/diskResource/client/events/selection/MetaTemplateInfoBtnSelected.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+import com.google.gwt.event.shared.EventHandler;
+import com.google.gwt.event.shared.GwtEvent;
+import com.google.gwt.event.shared.HandlerRegistration;
+
+/**
+ * A GWT event that fires when the user hits the "info" button for a metadata template
+ */
+public class MetaTemplateInfoBtnSelected
+        extends GwtEvent<MetaTemplateInfoBtnSelected.MetaTemplateInfoBtnSelectedHandler> {
```
Comment (tedgin) [inline de-lib/src/main/java/org/iplantc/de/diskResource/client/events/selection/MetaTemplateInfoBtnSelected.java]: Please add javadoc comments for all of the public interfaces.

File: de-lib/src/main/java/org/iplantc/de/diskResource/client/views/metadata/SelectMetadataTemplateView.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+import com.sencha.gxt.widget.core.client.container.VerticalLayoutContainer;
+import com.sencha.gxt.widget.core.client.grid.ColumnConfig;
+import com.sencha.gxt.widget.core.client.grid.ColumnModel;
+import com.sencha.gxt.widget.core.client.grid.Grid;
+import com.sencha.gxt.widget.core.client.selection.SelectionChangedEvent;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class SelectMetadataTemplateView extends Composite implements SelectionChangedEvent.HasSelectionChangedHandlers<MetadataTemplateInfo> {
```
Comment (tedgin) [inline de-lib/src/main/java/org/iplantc/de/diskResource/client/views/metadata/SelectMetadataTemplateView.java]: Please add javadoc comments for this class.

File: de-lib/src/main/java/org/iplantc/de/diskResource/client/views/metadata/dialogs/SelectMetadataTemplateDialog.java [hunk 4]
```diff
@@ -4,91 +4,43 @@
...
 import com.sencha.gxt.widget.core.client.selection.SelectionChangedEvent;
 
-import java.util.Arrays;
 import java.util.List;
 
 /**
  * Created by sriram on 5/4/16.
  */
-public class SelectMetadataTemplateDialog extends IPlantDialog implements IsWidget {
+public class SelectMetadataTemplateDialog extends IPlantDialog implements IsWidget,
```
Comment (tedgin) [inline de-lib/src/main/java/org/iplantc/de/diskResource/client/views/metadata/dialogs/SelectMetadataTemplateDialog.java]: Please update the javadoc with a description of the class.

File: de-lib/src/main/java/org/iplantc/de/diskResource/client/events/selection/MetadataInfoBtnSelected.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+import com.google.gwt.event.shared.EventHandler;
+import com.google.gwt.event.shared.GwtEvent;
+import com.google.gwt.event.shared.HandlerRegistration;
+
+/**
+ * A GWT event that fires when the user hits the "info" button for a metadata template
+ */
+public class MetadataInfoBtnSelected
+        extends GwtEvent<MetadataInfoBtnSelected.MetadataInfoBtnSelectedHandler> {
```
Comment (tedgin) [inline de-lib/src/main/java/org/iplantc/de/diskResource/client/events/selection/MetadataInfoBtnSelected.java]: Please javadoc comment on public inner interfaces.

File: de-lib/src/main/java/org/iplantc/de/diskResource/client/presenters/metadata/MetadataPresenterImpl.java [hunk 6]
```diff
@@ -49,7 +51,8 @@
 /**
  * @author jstroot sriram
  */
```
Comment (tedgin) [inline de-lib/src/main/java/org/iplantc/de/diskResource/client/presenters/metadata/MetadataPresenterImpl.java]: Please update the javadoc comment with a description of the class.

File: de-lib/src/main/java/org/iplantc/de/diskResource/client/views/metadata/SelectMetadataTemplateView.java [hunk 7]
```diff
@@ -28,7 +29,8 @@
 import java.util.Arrays;
 import java.util.List;
```
Comment (tedgin) [inline de-lib/src/main/java/org/iplantc/de/diskResource/client/views/metadata/SelectMetadataTemplateView.java]: Please add a javadoc comment for this class.

File: de-lib/src/main/java/org/iplantc/de/diskResource/client/views/metadata/cells/TemplateInfoCell.java [hunk 8]
```diff
@@ -13,18 +14,18 @@
...
 import com.google.gwt.user.client.Event;
-import com.google.gwt.user.client.rpc.AsyncCallback;
-import com.google.gwt.user.client.ui.HTML;
 import com.google.inject.Inject;
 
 import com.sencha.gxt.widget.core.client.Dialog;
 
 /**
  * Created by sriram on 7/7/16.
  */
```
Comment (tedgin) [inline de-lib/src/main/java/org/iplantc/de/diskResource/client/views/metadata/cells/TemplateInfoCell.java]: Please update the javadoc comment with a description of the class.

File: de-lib/src/main/java/org/iplantc/de/diskResource/client/views/metadata/dialogs/MetadataTemplateDescDlg.java [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+    @Inject
+    public MetadataTemplateDescDlg(TemplateInfoCell.TemplateInfoCellAppearance appearance) {
+        this.appearance = appearance;
+        setSize(appearance.descriptionWidth(), appearance.descriptionHeight());
+        setHideOnButtonClick(true);
+        setHeading(appearance.description());
+        setPredefinedButtons(PredefinedButton.CANCEL);
+    }
+
+    public void show(MetadataTemplateInfo info) {
```
Comment (tedgin) [inline de-lib/src/main/java/org/iplantc/de/diskResource/client/views/metadata/dialogs/MetadataTemplateDescDlg.java]: Please add a javadoc comment for this method.

File: de-lib/src/main/java/org/iplantc/de/diskResource/client/events/selection/MetaTemplateInfoBtnSelected.java (hunk 1 above)
Comment (ash3rz) [reply de-lib/src/main/java/org/iplantc/de/diskResource/client/events/selection/MetaTemplateInfoBtnSelected.java]: Yes - I use IntelliJ to automatically create these GWT events for me and it adds in `static` for some reason.  I'll clean those up.

Comment (tedgin) [reply de-lib/src/main/java/org/iplantc/de/diskResource/client/events/selection/MetaTemplateInfoBtnSelected.java]: There's no need to.  I shouldn't have asked that question here.  I'm sorry about that.

Comment (ash3rz) [reply de-lib/src/main/java/org/iplantc/de/diskResource/client/events/selection/MetaTemplateInfoBtnSelected.java]: No worries, it's a valid question!

**Comment by ash3rz** [comment]
Thanks for the review!

**Pull request merged**